src/TransformerGo.cs

```diff
@@ -75,54 +67,48 @@ private void TransformEnumTypes(CodeModelGo cmg)
             }
 
             // Add discriminators
-            foreach (var mt in cmg.ModelTypes)
+            foreach (var mt in cmg.ModelTypes.Cast<CompositeTypeGo>())

```

Comment:
Given the complexity of this algorithm it would be great to add a thorough comment explaining its intent.